core/src/splitting/RegressionSplittingRule.cpp

```diff
@@ -142,36 +160,55 @@ void RegressionSplittingRule::find_best_split_value_small_q(const Data& data,
   double sum_left = 0;
 
   // Compute decrease of impurity for each possible split
-  for (size_t i = 0; i < num_splits; ++i) {
-    n_left += counter[i];
-    weight_sum_left += weight_sums[i];
-    sum_left += sums[i];
-
-    // Skip this split if one child is too small.
-    if (n_left < min_child_size) {
-      continue;
-    }
-
-    // Stop if the right child is too small.
-    size_t n_right = size_node - n_left;
-    if (n_right < min_child_size) {
+  for (int j = 0; j < 2; ++j) {
```

Comment:
This is a great strategy, but for me the notation was a hard to follow. Here is a suggestion for how to clarify it (assuming it doesn't cause a perf regression):

```
  // Compute decrease of impurity for each possible split
  for (bool send_left : {false, true}) {
    if (send_left)  {
      // A normal split with no NAs, so we can stop early.
      if (n_missing == 0) {
        break;
      }
      
      // It is not necessary to adjust n_right or sum_right as the the missing
      // part is included in the total sum.
      n_left = n_missing;
      weight_sum_left = weight_sum_missing;
      sum_left = sum_missing;
    }
  ....
```